### DISCUSSION TITLE
How can i set the delete and search through limit in textchannel.purge() ?

### DISCUSSION THREAD

**Discussion by yannic-md** [Rapptz/discord.py#9246] [Q&A]
So I'm using `channel.purge(limit=10, check=check)` and my check is that:
```py
def check (m):
  return m.embeds
```

But that only searches in 10 messages, but I want to delete 10 messages that fulfill my check. There should be no limit to searching through - How can I do that and what would be the best way for it?

**Comment by du-cki** [comment]
Hello. There is no "native" way of doing this, you'll have better luck iterating through history until your quote gets met, wouldn't recommend this due to it being subjected to ratelimits.

**Reply by yannic-md** [reply]
Can't i set a count for the check or something if the limit is None?

**Reply by Vexs** [reply]
The problem arises where you have say, one million messages in a channel, and you want to delete the last 100 sent by someone who has only sent 10- naively, you'd search through the entire one million messages. 

That's why it's implemented the way it is, and why implementations like what you want are left up to the developer- that said, I wouldn't list ratelimits as a substantial worry unless you run into the situation listed above.

**Reply by yannic-md** [reply]
Okay but in this case, wouldn't it make sense to make a limit like the last 5,000 messages or something? Because I want to add a filter in my bot to remove only messages that have example attachments, or only mentions, or something like that. But I understand the struggle and the thoughts about that.

If a channel has one million messages, the author from the clear command probably wants to go through the last few days or hours. Maybe the higher the number of message in the chat, the smaller the search radius or something?

**Reply by yannic-md** [reply]
So, what would be a way to solve this issue if I would make a hard limit to search though 1.000 messages?

**Reply by du-cki** [reply]
Not recommending this but if you still want to do this, you can have a counter in the predicate and just stop deleting the embeds after `n` amount of times. This won't stop the iteration though.
